cts/cts-cli.in

```diff
@@ -2956,13 +2956,63 @@ class UpgradeRegressionTest(RegressionTest):
     <instance_attributes id-ref="mySmartFuse-outputpower-instanceparams"/>
   </primitive>
 </resources>
+"""
+        resource_cib2 = """
+<cib crm_feature_set="3.19.5" validate-with="pacemaker-1.3" epoch="2" num_updates="0" admin_epoch="0">
+  <configuration>
+    <crm_config>
+      <cluster_property_set id="cib-bootstrap-options">
+        <nvpair id="cib-bootstrap-options-stonith-enabled" name="stonith-enabled" value="false"/>
+      </cluster_property_set>
+    </crm_config>
+    <nodes/>
+    <resources/>
+    <constraints/>
+    <rsc_defaults>
+      <meta_attributes id="build-resource-defaults">
+        <nvpair id="build-resource-stickiness" name="resource-stickiness" value="1"/>
+      </meta_attributes>
+    </rsc_defaults>
+    <acls>
+      <acl_role id="acl-role"/>
+      <acl_user id="acl-user">
+        <role_ref id="acl-role"/>
+      </acl_user>
+    </acls>
+  </configuration>
+  <status/>
+</cib>
 """
 
+        t882_tests3 = [
+            Test("Run cibadmin with upgrade and force flags", "cibadmin -u --force",
+                 update_cib=True),
+        ]
+
+        t882_tests0 = [
+            Test("Run cibadmin with upgrade and force flags", "cibadmin -u --force",
+                 update_cib=True),
+        ]
+
+        t882_tests1 = [
+            Test("Run cibadmin with upgrade and force flags", "cibadmin -u --force -x '{cts_cli_data}/cibadmin_invalid_schema.xml'",
+                 update_cib=True),
+        ]
+
+        t882_tests2 = [
+            Test("Run cibadmin with upgrade and force flags", "cibadmin -u --force -p",
+                 update_cib=True, stdin=resource_cib2),
+        ]
+
         basic_tests = [
+
+            TestGroup(t882_tests3,
+                            env={"CIB_file": "{cts_cli_data}/cibadmin_invalid_schema.xml"}),
+
```

Comment:
You're not adding tests0, tests1, or tests2 to basic_tests so they will never be run.  I think you were just experimenting with different ways of providing the input for the tests, right?  In general, we're doing a pretty good job of testing that already so I think it's okay if you just use a single method here and remove all the others.  In that case, you can also get rid of resource_cib2 because it's just duplicating what's in the XML file you add in a later patch.